File: PAT (Basic-Level)/C/1037.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void swap(int *a, int *b);

int main()
{
    int gallonp, sicklep, knutp;
    int gallona, sicklea, knuta;
    int gallonr, sickler, knutr;
    int flat = 0;
    scanf("%d.%d.%d", &gallonp, &sicklep, &knutp);
    scanf("%d.%d.%d", &gallona, &sicklea, &knuta);

    if((gallona < gallonp) || (gallona == gallonp && (sicklea < sicklep)) || ((gallona == gallonp) && (sicklea == sicklep) && (knuta < knutp))){
        swap(&gallona, &gallonp);
        swap(&sicklea, &sicklep);
        swap(&knuta, &knutp);
        flat = 1;
    }

    if(knuta < knutp){
        knutr = knuta + 29 - knutp;
        sicklea--;
    }else{
        knutr = knuta - knutp;
    }

    if(sicklea < sicklep){
        sickler = sicklea + 17 - sicklep;
        gallona--;
    }else{
        sickler = sicklea - sicklep;
    }

    gallonr = gallona - gallonp;

    if(flat) printf("-%d.%d.%d", gallonr, sickler, knutr);
    else printf("%d.%d.%d", gallonr, sickler, knutr);
    return 0;
}

void swap(int *a, int *b){
    int temp;
    temp = *a;
    *a = *b;
    *b = temp;
}

File: PAT (Basic-Level)/Python/1037.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

import re

gallonp, sicklep, knutp, gallona, sicklea, knuta = list(map(int, re.split(' |\.', input())))
flat = 0

if (gallona < gallonp) or (gallona == gallonp and sicklea < sicklep) or (gallona == gallonp and sicklea == sicklep and knuta < knutp):
    gallona, gallonp = gallonp, gallona
    sicklea, sicklep = sicklep, sicklea
    knuta, knutp = knutp, knuta
    flat = 1

if knuta < knutp:
    knutr = knuta + 29 - knutp
    sicklea -= 1
else:
    knutr = knuta - knutp

if sicklea < sicklep:
    sickler = sicklea + 17 - sicklep
    gallona -= 1
else:
    sickler = sicklea - sicklep

gallonr = gallona - gallonp

if flat == 1:
    print("-%d.%d.%d" % (gallonr, sickler, knutr), end = '')
else:
    print("%d.%d.%d" % (gallonr, sickler, knutr), end = '')